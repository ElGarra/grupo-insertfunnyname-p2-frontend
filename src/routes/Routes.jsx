import React from 'react';
// import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Route } from 'react-router-dom';

import './Routes.scss';

import Footer from '../components/Footer/Footer';
import Navbar from '../components/Navbar/Navbar';
import Home from '../views/Home/Home';
import Login from '../views/Login/Login';
import Properties from '../views/Properties/Properties';
import Signup from '../views/Signup/Signup';
import AuthContextProvider from '../contexts/AuthContext';
import ReportContextProvider from '../contexts/ReportContext';
import Property from '../views/Property/Property';
import Profile from '../views/Profile/Profile';
import Meeting from '../views/Meeting/Meeting';
import Admin from '../views/Admin/Admin';

const Routes = () => (
  <AuthContextProvider>
    <ReportContextProvider>
      <Navbar />
      <div className="content">
        <div className="viewport">
          <Route path="/" exact component={Home} />
          <Route path="/login" exact component={Login} />
          <Route path="/signup" exact component={Signup} />
          <Route path="/properties" exact component={Properties} />
          <Route path="/properties/:propertyId" exact component={Property} />
          <Route path="/profile" exact component={Profile} />
          <Route path="/meetings/:meetingId" exact component={Meeting} />
          <Route path="/admin" exact component={Admin} />
        </div>
      </div>
      <Footer />
    </ReportContextProvider>
  </AuthContextProvider>
);

export default Routes;
