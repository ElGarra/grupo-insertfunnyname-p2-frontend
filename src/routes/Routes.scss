@import '../styles/colors.scss';

.content {
  display: flex;
  align-items: center;
  justify-items: center;
  justify-content: center;

  .viewport {
    padding: 20px;
    width: 66vw;
    min-width: 600px;
  }
}
