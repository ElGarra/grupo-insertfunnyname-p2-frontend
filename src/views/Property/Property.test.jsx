import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Routes from '../../routes/Routes';

const { act } = renderer;

function TestRouter({ path }) {
  return (
    <MemoryRouter initialEnteries={[path]}>
      <Routes />
    </MemoryRouter>
  );
}

TestRouter.propTypes = {
  path: PropTypes.string.isRequired,
};

const user = {
  firstName: 'John',
  lastName: 'Doe',
  email: 'johndoe@example.com',
  access_token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOjE1MiwiaWF0IjoxNjIzNDczOTMwfQ.zXFM-exk-29-afuxH7glrAmrxZokQeCZXd8r_efIbvg',
  token_type: 'Bearer',
};

const sessionExpiration = new Date(
  new Date().getTime() + 1000 * 60 * 60 * 24,
);

const localStorageMapping = {
  user,
  sessionExpiration,
};

describe('Properties', () => {
  let testRenderer;
  beforeAll(() => {
    act(() => {
      testRenderer = renderer.create(<TestRouter path="/property/1" />);
    });
  });
  describe('when user is not logged in', () => {
    it('renders the property page', () => {
      const tree = testRenderer.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('when user is logged in', () => {
    beforeEach(() => {
      global.Storage.prototype.getItem = jest.fn(
        (key) => JSON.stringify(localStorageMapping[key]),
      );
    });
    afterEach(() => {
      global.Storage.prototype.getItem.mockReset();
    });
    it('renders the property page', () => {
      const tree = testRenderer.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
