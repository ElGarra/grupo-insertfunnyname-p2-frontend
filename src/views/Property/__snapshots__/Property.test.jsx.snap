// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties when user is logged in renders the property page 1`] = `
Array [
  <header
    className="Navbar"
  >
    <nav>
      <div>
        <div
          className="brand"
        >
          <div
            className="logo"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="Logo"
                src="logo-brand.png"
              />
            </a>
          </div>
        </div>
      </div>
      <div>
        <div
          className="searchbar-container"
        >
          <form
            className="navbar-form"
          >
            <label
              htmlFor="search"
            >
              Search
              <input
                id="search"
                type="text"
              />
            </label>
          </form>
        </div>
      </div>
      <div>
        <div
          className="button-container"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            <button
              className="BaseButton"
              type="button"
            >
              Log In
            </button>
          </a>
        </div>
        <div
          className="button-container"
        >
          <a
            href="/signup"
            onClick={[Function]}
          >
            <button
              className="BaseButton"
              type="button"
            >
              Sign up
            </button>
          </a>
        </div>
      </div>
    </nav>
  </header>,
  <div
    className="navbar-offset"
  />,
  <div
    className="content"
  >
    <div
      className="viewport"
    >
      <div
        className="Home"
      >
        <div
          className="landing-image"
        >
          <img
            alt="Welcome to FindHomy"
            src="https://images.unsplash.com/photo-1524813686514-a57563d77965"
          />
        </div>
        <div
          className="access-button"
        >
          <h1>
            Welcome
          </h1>
          <a
            href="/properties"
            onClick={[Function]}
          >
            <button
              className="BaseButton"
              type="button"
            >
              Browse our properties
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="Footer"
  >
    <div
      id="logo-footer"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="wordmark"
        >
          FindHomy
        </h1>
        <img
          alt="Logo"
          src="logo512.png"
        />
      </a>
    </div>
    <div
      className="footer-box"
    >
      <ul>
        <li>
          <a
            href="https://www.youtube.com/watch?v=5E2dsqjoFXg&ab_channel=JohnTimston"
            rel="noreferrer"
            target="_blank"
          >
            About
          </a>
        </li>
        <hr />
        <li>
          <a
            href="https://www.youtube.com/watch?v=pCjBetzC8ME&ab_channel=Meow%3FWoof%21"
            rel="noreferrer"
            target="_blank"
          >
            Contact
          </a>
        </li>
        <hr />
        <li>
          <a
            href="https://www.youtube.com/watch?v=GfT5CqrWJW0&ab_channel=ALIBA"
            rel="noreferrer"
            target="_blank"
          >
            Terms of Service
          </a>
        </li>
      </ul>
    </div>
    <div
      className="footer-box"
    >
      <ul>
        <li>
          <a
            href="https://documenter.getpostman.com/view/15970655/TzeTKA1X"
            rel="noreferrer"
            target="_blank"
          >
            Docs
          </a>
        </li>
        <hr />
        <li>
          <a
            href="https://trippybook.herokuapp.com/posts"
            rel="noreferrer"
            target="_blank"
          >
            TrippyBook
          </a>
        </li>
      </ul>
    </div>
    <div
      className="copyright"
    >
      <span
        className="subtitle2"
      >
        © Copyright 2021 InsertFunnyName
      </span>
    </div>
  </footer>,
]
`;

exports[`Properties when user is not logged in renders the property page 1`] = `
Array [
  <header
    className="Navbar"
  >
    <nav>
      <div>
        <div
          className="brand"
        >
          <div
            className="logo"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="Logo"
                src="logo-brand.png"
              />
            </a>
          </div>
        </div>
      </div>
      <div>
        <div
          className="searchbar-container"
        >
          <form
            className="navbar-form"
          >
            <label
              htmlFor="search"
            >
              Search
              <input
                id="search"
                type="text"
              />
            </label>
          </form>
        </div>
      </div>
      <div>
        <div
          className="button-container"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            <button
              className="BaseButton"
              type="button"
            >
              Log In
            </button>
          </a>
        </div>
        <div
          className="button-container"
        >
          <a
            href="/signup"
            onClick={[Function]}
          >
            <button
              className="BaseButton"
              type="button"
            >
              Sign up
            </button>
          </a>
        </div>
      </div>
    </nav>
  </header>,
  <div
    className="navbar-offset"
  />,
  <div
    className="content"
  >
    <div
      className="viewport"
    >
      <div
        className="Home"
      >
        <div
          className="landing-image"
        >
          <img
            alt="Welcome to FindHomy"
            src="https://images.unsplash.com/photo-1524813686514-a57563d77965"
          />
        </div>
        <div
          className="access-button"
        >
          <h1>
            Welcome
          </h1>
          <a
            href="/properties"
            onClick={[Function]}
          >
            <button
              className="BaseButton"
              type="button"
            >
              Browse our properties
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="Footer"
  >
    <div
      id="logo-footer"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="wordmark"
        >
          FindHomy
        </h1>
        <img
          alt="Logo"
          src="logo512.png"
        />
      </a>
    </div>
    <div
      className="footer-box"
    >
      <ul>
        <li>
          <a
            href="https://www.youtube.com/watch?v=5E2dsqjoFXg&ab_channel=JohnTimston"
            rel="noreferrer"
            target="_blank"
          >
            About
          </a>
        </li>
        <hr />
        <li>
          <a
            href="https://www.youtube.com/watch?v=pCjBetzC8ME&ab_channel=Meow%3FWoof%21"
            rel="noreferrer"
            target="_blank"
          >
            Contact
          </a>
        </li>
        <hr />
        <li>
          <a
            href="https://www.youtube.com/watch?v=GfT5CqrWJW0&ab_channel=ALIBA"
            rel="noreferrer"
            target="_blank"
          >
            Terms of Service
          </a>
        </li>
      </ul>
    </div>
    <div
      className="footer-box"
    >
      <ul>
        <li>
          <a
            href="https://documenter.getpostman.com/view/15970655/TzeTKA1X"
            rel="noreferrer"
            target="_blank"
          >
            Docs
          </a>
        </li>
        <hr />
        <li>
          <a
            href="https://trippybook.herokuapp.com/posts"
            rel="noreferrer"
            target="_blank"
          >
            TrippyBook
          </a>
        </li>
      </ul>
    </div>
    <div
      className="copyright"
    >
      <span
        className="subtitle2"
      >
        © Copyright 2021 InsertFunnyName
      </span>
    </div>
  </footer>,
]
`;
